File: 45secondapps_ui/src/services/FeatureService.js
import http from "../http-common"; 

class FeatureService {
  getAllFeatures(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/feature/features`, searchDTO);
  }

  get(featureId) {
    return this.getRequest(`/feature/${featureId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/feature?field=${matchData}`, null);
  }

  addFeature(data) {
    return http.post("/feature/addFeature", data);
  }

  update(data) {
  	return http.post("/feature/updateFeature", data);
  }
  
  uploadImage(data,featureId) {
  	return http.postForm("/feature/uploadImage/"+featureId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FeatureService();

File: src/main/java/com/45secondapps/service/FeatureService.java
package com.45secondapps.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.45secondapps.domain.Feature;
import com.45secondapps.dto.FeatureDTO;
import com.45secondapps.dto.FeatureSearchDTO;
import com.45secondapps.dto.FeaturePageDTO;
import com.45secondapps.dto.FeatureConvertCriteriaDTO;
import com.45secondapps.service.GenericService;
import com.45secondapps.dto.common.RequestDTO;
import com.45secondapps.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FeatureService extends GenericService<Feature, Integer> {

	List<Feature> findAll();

	ResultDTO addFeature(FeatureDTO featureDTO, RequestDTO requestDTO);

	ResultDTO updateFeature(FeatureDTO featureDTO, RequestDTO requestDTO);

    Page<Feature> getAllFeatures(Pageable pageable);

    Page<Feature> getAllFeatures(Specification<Feature> spec, Pageable pageable);

	ResponseEntity<FeaturePageDTO> getFeatures(FeatureSearchDTO featureSearchDTO);
	
	List<FeatureDTO> convertFeaturesToFeatureDTOs(List<Feature> features, FeatureConvertCriteriaDTO convertCriteria);

	FeatureDTO getFeatureDTOById(Integer featureId);







}





